### ISSUE TITLE
Support `pull_request_target` and `pull_request_review`

### ISSUE THREAD

**Issue by ghengeveld** [chromaui/chromatic-cli#196]
By @NicholasBoll in https://github.com/chromaui/action/issues/56:

Github Actions now support pull requests from forks: https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/

More info about the `pull_request_target` event type: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target

However, the code explicitly prevents that type: https://github.com/chromaui/action/blob/c427f36903fb8c899e440bc90cb51822ef2959ed/src/main.ts#L25-L55

Here's an example run failure: https://github.com/Workday/canvas-kit/pull/788/checks?check_run_id=1114827451

`pull_request_target` should be a valid event.

**Note**: from our testing, adding `pull_request_target` or changing `pull_request` to `pull_request_target` in a Pull Request will result in Github Actions not running for that pull request. The event type is supposed to run code from the base branch (the branch the PR is being merged into) which gets Actions into a state where nothing triggers.

---

By @marboro92 in https://github.com/chromaui/action/issues/59:

My team would like to run chromatic checks only after a review is submitted, but currently the pull_request_review event (info on this event: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_review) is not supported by chromaui. 

Would it be possible to add support for pull_request_review?

**Comment by tmeasday** [comment]
@ndelangen can you tell a bit more about what's happening in this ticket? Do you understand what these events mean/when they trigger?

**Comment by ndelangen** [comment]
@tmeasday not yet, but I'll put this on my todo list to find out and assess if it's ok to just add these events 👍

**Comment by ndelangen** [comment]
@ghengeveld can you do a release this week?

**Comment by penx** [comment]
From the linked blog post

> However, instead of running against the workflow and code from the merge commit, the event runs against the workflow and code from the base of the pull request

From what I've seen, the Chromatic action will publish and compare the storybook from the base branch rather than with the proposed changes, resulting in no detected changes.

https://github.com/govuk-react/govuk-react/pull/1077#discussion_r862464284

Any idea if there's a way to use the pull_request_target but still get the visual diff to include the proposed changes?

**Comment by penx** [comment]
Ah perhaps this blog post contains the answer - it is in relation to dependabot, but I think the same approach would work

https://blog.alphasmanifesto.com/2021/11/07/yarn-2-dependabot/


```yml
name: Dependabot
on:
  - pull_request_target

jobs:
  fix-lockfile:
    runs-on: ubuntu-latest
    if: |
      github.actor == 'dependabot[bot]' &&
      contains(github.event.pull_request.head.ref, 'dependabot/npm_and_yarn/')

    # IMPORTANT: setting YARN_ENABLE_SCRIPTS=false is critical to ensure that untrusted
    # PRs can't add an npm package and then use that to execute untrusted code in
    # a trusted context. See links at the top of this workflow for further details.
    # See also: https://github.com/yarnpkg/berry/issues/1679#issuecomment-669937860
    env:
      YARN_ENABLE_SCRIPTS: false
      YARN_ENABLE_IMMUTABLE_INSTALLS: false

    steps:
      - uses: actions/checkout@v2.3.5
        with:
          # Using a Personal Access Token here is required to trigger workflows on our new commit.
          # The default GitHub token doesn't trigger any workflows.
          # See: https://github.community/t/push-from-action-does-not-trigger-subsequent-action/16854/2
          token: ${{ secrets.DEPENDABOT_PAT }}
          ref: ${{ github.event.pull_request.head.ref }}
```

i.e. 

1. trigger the build via pull_request_target
2. make any security changes needed to ensure 3rd party code is not executed
3. use actions/checkout to switch branches
4. install dependencies (securely) and execute chromatic github action

**Comment by penx** [comment]
this is also covered here https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

**Comment by penx** [comment]
I think the basic setup would look something like this

```yml
name: 'Chromatic'
on:
  pull_request_target:
    branches:
      - main
      - master
jobs:
  chromatic-deployment:
    runs-on: ubuntu-latest
    env:
      YARN_ENABLE_SCRIPTS: false
      YARN_ENABLE_IMMUTABLE_INSTALLS: false
    steps:
    - uses: actions/checkout@v2
        with:
          ref: ${{ github.event.pull_request.head.ref }}
          fetch-depth: 0
    - name: Install dependencies
      run: yarn
    - name: Publish to Chromatic
      uses: chromaui/action@v1
      with:
        projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
```
